src/SourceLocator/Type/FileSystemIteratorSourceLocator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace BetterReflection\SourceLocator\Type;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+
+/**
+ * This source locator loads all php files from \FileSystemIterator
+ */
+class FileSystemIteratorSourceLocator implements SourceLocator
+{
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $_aggregatedSourceLocator;
+
+    /**
+     * @var \FilesystemIterator
+     */
+    private $fileSystemIterator;
+
+    public function __construct(\FilesystemIterator $filesystemIterator)
```

Comment:
The reason for not type-hinting against `FilesystemIterator` is that it is a concrete class, btw.